tensorstore/kvstore/s3/credentials/aws_credential_provider.h

```diff
@@ -48,45 +45,19 @@ struct AwsCredentials {
 
 /// Base class for S3 Credential Providers
 ///
-/// Implementers should override GetCredentials
+/// Implementers should override GetCredentials, IsExpired and,
+/// if the credential source supports the
+/// representation of an expiry date, ExpiresAt.
 class AwsCredentialProvider {
- public:
-  virtual ~AwsCredentialProvider() = default;
-  virtual Result<AwsCredentials> GetCredentials() = 0;
+  public:
+    virtual ~AwsCredentialProvider() = default;
+    virtual Result<AwsCredentials> GetCredentials() = 0;
+    virtual bool IsExpired() = 0;
```

Comment:
Why have an IsExpired() and an ExpiresAt()? stick to a single thing.

Reply:
Just for context: The original intent of this PR was to align the implementation with the aws go sdk, which had this sort of interface. This isn't relevant now that we're going with `DefaultCredentialsProvider`,